base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
- [ ] OPTIONAL
It's a best practice to mark `nameable` with `attr_accessor`, to give an indication that we intend to both _read_ and write this variable. Let's endeavor to improve upon the same.